daemon/config/config_test.go

```diff
@@ -80,3 +252,54 @@ func TestGetConflictConfigurations(t *testing.T) {
 	assert.Equal("found conflict flags in command line and config file: from flag: a1 and from config file: a1",
 		getConflictConfigurations(flags, fileflags).Error())
 }
+
+func TestMergeConfigurations(t *testing.T) {
+	assert := assert.New(t)
+	flags := pflag.NewFlagSet("cmflags", pflag.ContinueOnError)
+
+	// Test configuration file doesn't exist
+	cfg := &Config{
+		ConfigFile: "/tmp/non-existent.json",
+	}
+
+	assert.Equal(nil, cfg.MergeConfigurations(flags))
+
+	// Test invalid content format
+	configFile := "/tmp/test-merge-config-1.json"
+	err := ioutil.WriteFile(configFile, []byte(""), 0644)
+	assert.Equal(nil, err)
+
+	cfg = &Config{
+		ConfigFile: configFile,
+	}
+
+        err = cfg.MergeConfigurations(flags)
```

Comment:
I think you do not the `gofmt` this place?
I did not check everywhere, while I think it is easy to use `gomft` to format all the file. @chuanchang

Reply:
sure, I did gofmt for v1 patch, but forgot it for v2, thanks for pointing it out.